src/Rules/Comparison/ConstantConditionRuleHelper.php

```diff
@@ -49,17 +46,6 @@ public function shouldSkip(Scope $scope, Expr $expr): bool
 			return true;
 		}
 
-		if (
-			$expr instanceof FuncCall
-			|| $expr instanceof MethodCall
-			|| $expr instanceof Expr\StaticCall
-		) {
-			$isAlways = $this->impossibleCheckTypeHelper->findSpecifiedType($scope, $expr);
```

Comment:
I think this should stay here. We don't want a new error to be reported here: https://github.com/phpstan/phpstan/blob/b127a70bcc46ec3be6a5439771bf6539e189e3ad/e2e/php8/test.php#L11 by BooleanNotConstantConditionRule. It's already taken care of by ImpossibleCheckTypeFunctionCallRule.

Reply:
yeah I realized the same too late :) will bring it back and fix whatever other error it caused (see https://github.com/phpstan/phpstan-src/pull/1254#discussion_r859806273)